plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -26,21 +26,24 @@
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.lang.Integer.parseInt;
 import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
 
 public final class PartitionFields
 {
-    private static final String NAME = "[a-z_][a-z0-9_]*";
-    private static final String FUNCTION_ARGUMENT_NAME = "\\((" + NAME + ")\\)";
-    private static final String FUNCTION_ARGUMENT_NAME_AND_INT = "\\((" + NAME + "), *(\\d+)\\)";
-
-    private static final Pattern IDENTITY_PATTERN = Pattern.compile(NAME);
-    private static final Pattern YEAR_PATTERN = Pattern.compile("year" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern MONTH_PATTERN = Pattern.compile("month" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern DAY_PATTERN = Pattern.compile("day" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern HOUR_PATTERN = Pattern.compile("hour" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern BUCKET_PATTERN = Pattern.compile("bucket" + FUNCTION_ARGUMENT_NAME_AND_INT);
-    private static final Pattern TRUNCATE_PATTERN = Pattern.compile("truncate" + FUNCTION_ARGUMENT_NAME_AND_INT);
-    private static final Pattern VOID_PATTERN = Pattern.compile("void" + FUNCTION_ARGUMENT_NAME);
+    private static final String UNQUOTED_IDENTIFIER = "[a-zA-Z_][a-zA-Z0-9_]*";
+    private static final String IDENTIFIER = "\\s*(" + UNQUOTED_IDENTIFIER + ")\\s*";
+
+    private static final String FUNCTION_ARGUMENT_NAME = "\\(" + IDENTIFIER + "\\)\\s*";
+    private static final String FUNCTION_ARGUMENT_NAME_AND_INT = "\\(" + IDENTIFIER + ",\\s*(\\d+)\\s*\\)\\s*";
+
+    private static final Pattern IDENTITY_PATTERN = Pattern.compile(IDENTIFIER);
+    private static final Pattern YEAR_PATTERN = Pattern.compile("\\s*(?i:year)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern MONTH_PATTERN = Pattern.compile("\\s*(?i:month)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern DAY_PATTERN = Pattern.compile("\\s*(?i:day)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern HOUR_PATTERN = Pattern.compile("\\s*(?i:hour)" + FUNCTION_ARGUMENT_NAME);
```

Comment:
I understand why "Move lowercasing to PartitionFields" adds changes `hour` to `(?i:hour)`, but why is `\s*` added too?

Reply:
not sure, but was it to support the case to "not to fail with whitespaces"
```
truncate(name, 5),    hour(dt)
```
while adding `sorted_by`, also I'm handing white spaces and then trimming, for the cases where by mistake white spaces are added.

Something like this:

```
    private static final Pattern YEAR_PATTERN_ASC_NULLS_FIRST = Pattern.compile("\\s*?(?i:year)" + FUNCTION_ARGUMENT_NAME + "\\s*?(ASC)\\s*?(NULLS\\s*?FIRST)\\s*?");
```

Though definitely, we can discuss in detail the sorted_by PR, it was relevant to bring it up.

Reply:
It was indeed to support arbitrary whitespace and do parsing as lenient as possible. See also the remark on `TestPartitionFields`.

> Why would we want to add support for surrounding spaces? looks like redundant flexibility to me.

Reply:
I see value in supporting both `truncate(comment,⎵13)` and `truncate(comment,13)`
i understand that being whitespace agnostic leads us to consequently accept `truncate(comment,13⎵⎵)` and `⎵⎵truncate(comment,13)⎵⎵`.

Let's remove it from this PR -- as it's not really a related change -- and revisit separately..